db/migrations/2003110348_extract_results_from_attempts.sql

```diff
@@ -0,0 +1,423 @@
+-- +migrate Up
+RENAME TABLE `attempts` TO `results`;
+
+ALTER TABLE `results`
+    DROP FOREIGN KEY `fk_attempts_creator_id_users_group_id`,
+    DROP CHECK `cs_attempts_order`,
+    DROP CHECK `cs_attempts_score_computed_is_valid`,
+    ADD CONSTRAINT `cs_results_score_computed_is_valid` CHECK ((`score_computed` between 0 and 100)),
+    DROP CHECK `cs_attempts_score_edit_value_is_valid`,
+    ADD CONSTRAINT `cs_results_score_edit_value_is_valid` CHECK ((ifnull(`score_edit_value`,0) between -(100) and 100)),
+    RENAME COLUMN `group_id` TO `participant_id`,
+    RENAME INDEX `group_item` TO `participant_id_item_id`,
+    RENAME INDEX `group_item_score_desc_score_obtained_at` TO `participant_id_item_id_score_desc_score_obtained_at`,
+    ADD COLUMN `attempt_id` BIGINT(20) NOT NULL DEFAULT 0 AFTER `participant_id`;
+
+DROP TRIGGER `before_insert_attempts`;
+
+CREATE TABLE `attempts` (
+    `participant_id` BIGINT(20) NOT NULL,
+    CONSTRAINT `fk_attempts_participant_id_groups_id` FOREIGN KEY (`participant_id`) REFERENCES `groups`(`id`) ON DELETE CASCADE,
+    `id` BIGINT(20) NOT NULL
+        COMMENT 'Identifier of this attempt for this participant, 0 is the default attempt for the participant, the next ones are sequentially assigned.',
+    PRIMARY KEY (`participant_id`, `id`),
+    `creator_id` BIGINT(20) DEFAULT NULL COMMENT 'The user who created this attempt',
+    CONSTRAINT `fk_attempts_creator_id_users_group_id` FOREIGN KEY (`creator_id`) REFERENCES `users`(`group_id`) ON DELETE SET NULL,
+    `parent_attempt_id` BIGINT(20) DEFAULT NULL COMMENT 'The attempt from which this one was forked. NULL for the default attempt.',
+    `root_item_id` BIGINT(20) DEFAULT NULL COMMENT 'The item on which the attempt was created',
+    CONSTRAINT `fk_attempts_root_item_id_items_id`FOREIGN KEY (`root_item_id`) REFERENCES `items`(`id`) ON DELETE SET NULL,
+    `created_at` DATETIME NOT NULL DEFAULT NOW()
+        COMMENT 'Time at which the attempt was manually created or was first marked as started (should be when it is first visited).',
+    INDEX `participant_id_parent_attempt_id_root_item_id` (`participant_id`, `parent_attempt_id`, `root_item_id`)
+    -- We cannot add the following constraint because it would set both `participant_id` and `parent_attempt_id` to NULL
+    -- on deletion of the parent attempt.
+    -- CONSTRAINT `attempts_participant_id_parent_attempt_id_attempts_participant_id_id`
+    --    FOREIGN KEY (`participant_id`, `parent_attempt_id`) REFERENCES `attempts`(`participant_id`, `id`) SET NULL,
+) ENGINE=InnoDB DEFAULT CHARSET=utf8
+    COMMENT 'Attempts of participants (team or user) to solve a subtree of items. An attempt may have several answers. Every participant has a default attempt.';
```

Comment:
"An attempt may have several answers."- > "An attempt may have several answers for a same item."

Reply:
fixed